src/ophys_etl/modules/segmentation/graph_utils/edge_attributes.py

```diff
@@ -0,0 +1,381 @@
+        (default = False)
+
+    Returns
+    -------
+    pearsons: np.ndarray
+        The result of running cdist on the input traces masked according
+        to the specified filter_fraction
+    """
+    discard = 1.0-filter_fraction
+    thresh0 = np.quantile(node0, discard)
+    mask0 = np.where(node0 >= thresh0)[0]
+    thresh1 = np.quantile(node1, discard)
+    mask1 = np.where(node1 >= thresh1)[0]
+
+    global_mask = [mask0, mask1]
+    if full_neighborhood:
+        for trace in neighbor_traces:
+            thresh = np.quantile(trace, discard)
+            mask = np.where(trace >= thresh)[0]
+            global_mask.append(mask)
+    global_mask = np.unique(np.concatenate(global_mask))
+
+    masked_node0 = node0[global_mask]
+    masked_node1 = node1[global_mask]
+    masked_neighbors = []
+    for trace in neighbor_traces:
+        masked_neighbors.append(trace[global_mask])
+
+    pearsons = 1.0 - cdist([masked_node0, masked_node1],
+                           masked_neighbors,
+                           metric="correlation")
+
+    return pearsons
+
+
+def add_hnc_gaussian_metric(graph: nx.Graph,
+                            video_path: Path,
+                            neighborhood_radius: Union[int, float],
+                            attribute_name: str = "hnc_Gaussian",
+                            filter_fraction: Optional[float] = None,
+                            full_neighborhood: bool = False) -> nx.Graph():
+    """for each pair of pixels represented by an edge, calculates the Gaussian
+    similarity in correlation space to a neighborhood of size
+    'neighborhood_radius' where the correlation coefficient of all pairs of
+    pixels in that neighborhood are considered.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        a graph with nodes like (row, col) and edges connecting them
+    video_path: Path
+        path to an hdf5 video file, assumed to have a dataset "data"
+        nframes x nrow x ncol
+    neighborhood_radius: Union[int, float]
+        pixels within <= this radius of the pair centroid are considered
+        in the neighborhood.
+    attribute_name: str
+        name set on each edge for this calculated value
+    filter_fraction: Optional[float]
+        If not None, take the brightest filter_fraction pixels when
+        calculating the Pearson correlation within a neighborhood.
+    full_neighborhood: bool
+        If True, use all of the pixels in the neighborhood when selecting
+        the timesteps to correlate. When False, only use node0 and node1
+        (default = False)
+
+    Returns
+    -------
+    new_graph: nx.Graph
+        an undirected networkx graph, with attribute added to edges
+
+    Notes
+    -----
+    See:
+    "HNCcorr: A Novel Combinatorial Approach for Cell Identification
+    in Calcium-Imaging Movies"
+    Spaen et. al.
+    https://pubmed.ncbi.nlm.nih.gov/31058211/
+
+    """
+
+    with h5py.File(video_path, "r") as f:
```

Comment:
Would be better for this function to take `nrow`, `ncol` params rather than loading in an h5.